api/client/container/run.go

```diff
@@ -262,10 +262,15 @@ func runRun(dockerCli *client.DockerCli, flags *pflag.FlagSet, opts *runOptions,
 		return nil
 	}
 
-	status := <-statusChan
-	if status != 0 {
-		return cli.StatusError{StatusCode: status}
+	select {
+	case status := <-statusChan:
+		if status != 0 {
+			return cli.StatusError{StatusCode: status}
+		}
+	case err := <-errChan:
+		return fmt.Errorf("Error waiting container's exit code: %v", err)
```

Comment:
"Error waiting for..."?

Also, `error` string should generally start with a lowercase character.